storage/imx6ull-flash/flashsrv.c

```diff
@@ -275,7 +275,7 @@ static int flashsrv_partoff(id_t id, int raw, size_t start, size_t size, size_t
 		return -ENOENT;
 
 	TRACE("Partition: size %u, start %u (in EB)", p->size, p->start);
-	if ((start + size) > (p->size * erase_block_size))
+	if (start + size > (p->start + p->size) * erase_block_size)
```

Comment:
I don't believe this is valid.

the `start`, `size` params are within single partition, so they indeed should be within <0, p->size * erase_block_size>.

Maybe this function is being used wrongly somewhere?